src/Network/HTTP/Download/Verified.hs

```diff
@@ -179,21 +179,37 @@ hashChecksToZipSink :: MonadThrow m => Request -> [HashCheck] -> ZipSink ByteStr
 hashChecksToZipSink req = traverse_ (ZipSink . sinkCheckHash req)
 
 -- 'Control.Retry.recovering' customized for HTTP failures
-recoveringHttp :: MonadUnliftIO m
+recoveringHttp :: (MonadUnliftIO m, MonadLogger m)
                => RetryPolicy -> m a -> m a
 recoveringHttp retryPolicy =
 #if MIN_VERSION_retry(0,7,0)
-    helper $ recovering retryPolicy handlers . const
+    helper $ \run -> recovering retryPolicy (handlers run) . const
 #else
-    helper $ recovering retryPolicy handlers
+    helper $ \run -> recovering retryPolicy (handlers run)
 #endif
   where
-    helper wrapper action = withRunInIO $ \run -> wrapper (run action)
-
-    handlers = [const $ Handler alwaysRetryHttp,const $ Handler retrySomeIO]
-
-    alwaysRetryHttp :: Monad m => HttpException -> m Bool
-    alwaysRetryHttp _ = return True
+    helper :: MonadUnliftIO m => (UnliftIO m -> IO a -> IO a) -> m a -> m a
+    helper wrapper action = withUnliftIO $ \run -> wrapper run (unliftIO run action)
+
+    handlers :: MonadLogger m => UnliftIO m -> [RetryStatus -> Handler IO Bool]
+    handlers run = [Handler . alwaysRetryHttp (unliftIO run),const $ Handler retrySomeIO]
+
+    alwaysRetryHttp :: (MonadLogger m', Monad m) => (m' () -> m ()) -> RetryStatus -> HttpException -> m Bool
+    alwaysRetryHttp run rs e = do
+      run $ do
+        logWarn $ Text.unwords
+          [ "Retry number"
+          , Text.pack $ show (rsIterNumber rs)
+          , "after a total delay of"
+          , Text.pack $ show (rsCumulativeDelay rs)
+          , "us"
+          ]
+        logWarn $ Text.unwords
+          [ "If you see this warning and stack fails,"
```

Comment:
Would be good to narrow "fails" a bit, perhaps "If you see this warning, and this download doesn't succeed,"